hpy/tools/autogen/public_api.h

```diff
@@ -307,6 +307,56 @@ HPy HPy_Call(HPyContext *ctx, HPy callable, const HPy *args, size_t nargs, HPy k
 HPy_ID(262)
 HPy HPy_CallMethod(HPyContext *ctx, HPy name, const HPy *args, size_t nargs, HPy kwnames);
 
+/**
+ * Return a new iterator for iterable object ``obj``. This is the equivalent
+ * of the Python expression ``iter(obj)``.
+ *
+ * :param ctx:
+ *     The execution context.
+ * :param obj:
+ *     An iterable Python object (must not be ``HPy_NULL``). If the object is
+ *     not iterable, a ``TypeError`` will be raised.
+ *
+ * :returns:
+ *     The new iterator, ``obj`` itself if it is already an iterator, or
+ *     ``HPy_NULL`` on failure.
+ */
+HPy_ID(269)
+HPy HPy_GetIter(HPyContext *ctx, HPy obj);
+
+/**
+ * Return the next value from iterator ``obj``.
+ *
+ * :param ctx:
+ *     The execution context.
+ * :param obj:
+ *     An iterator Python object (must not be ``HPy_NULL``). This can be 
+ *     verified with ``HPy_IterCheck``. Otherwise, the behavior is undefined
+ *     (verification of the argument is only done in debug mode).
```

Comment:
> verification of the argument is only done in debug mode

minor: I think this is not the case yet? If you want to add that take a look at other functions that the debug mode "overrides" to add such checks, but it's fine as-is, we can add it later.

Reply:
My bad. I was lazy in adding that, having copy-pasted another docstring as a template and forgetting to actually verify that bit. I'll remove the parenthesized clause.